gpytorch/utils/stochastic_lq.py

```diff
@@ -64,7 +65,7 @@ def evaluate(self, t_mats, matrix_size, eigenvalues, eigenvectors, funcs):
             )
 
         batch_size = t_mats.size(1)
-        results = [t_mats.new(batch_size).zero_()] * len(funcs)
+        results = [torch.zeros(batch_size, dtype=t_mats.dtype, device=t_mats.device)] * len(funcs)
```

Comment:
This DOES NOT create `len(funcs)` individual tensors. Instead, we're referencing the same tensor `len(funcs)` times. Use `[torch.zeros(batch_size, dtype=t_mats.dtype, device=t_mats.device) for _ in len(funcs)]` instead.

Reply:
Fixed!